release-20.1: backupccl: correctly manage temp objects during a full cluster restore

adityamaru opened the pull request:
Discussion about the design at #56512.

This is a backport of #56956.

Fixes: #56512

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/61735)
<!-- Reviewable:end -->

adityamaru committed:
backupccl: correctly manage temp objects during a full cluster restore

Discussion about the design at #56512.

This is a backport of #56956.

Fixes: #56512

Release note: None
